[minhkiwi14] iP

minhkiwi14 opened the pull request:
# Kiwi

> "Focus on meaningful conversations, and let Kiwi handle the rest." – Anonymous

The chatbot Kiwi helps you improve your communication by making sure you remember important points in a conversation. It's:

- user-friendly
- simple to use
- **EXTREMELY** efficient

Here’s how you can get started:

1. Download Kiwi from [here](https://github.com/minhkiwi14/ip).
2. Run the application.
3. Start organizing your tasks.
4. Let Kiwi enhance your productivity! :kiwi_fruit:

And guess what? Kiwi is completely **FREE**!

### Features:

- [x] Managing conversations
- [x] Summarizing key points
- [ ] Scheduling reminders (coming soon)
- [ ] Sentiment analysis (coming soon)

If you're a Java developer, you can explore Kiwi’s code and improve it. Here’s a sample `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Kiwi.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

minhkiwi14 committed:
Rename main class

minhkiwi14 committed:
Change the logo

minhkiwi14 committed:
Add greeting and exiting

minhkiwi14 committed:
Update the project name

minhkiwi14 committed:
Add support for echo

minhkiwi14 committed:
Add indentation and horizontal lines

minhkiwi14 committed:
Add support for add and list

minhkiwi14 committed:
Add class Task

minhkiwi14 committed:
Add support for done and undone

minhkiwi14 committed:
Add class Todo, Deadline and Event

minhkiwi14 committed:
Add support for todo, deadline and event

minhkiwi14 committed:
Improve the printing

minhkiwi14 committed:
Add automated text UI testing

minhkiwi14 committed:
Add class for handling exceptions

minhkiwi14 committed:
Modularize the code for tasks

minhkiwi14 committed:
Handle two errors

minhkiwi14 committed:
Implement ArrayList

minhkiwi14 committed:
Add support for delete

minhkiwi14 committed:
Implement switch-case

minhkiwi14 committed:
Add getters

minhkiwi14 committed:
Add class Storage for saving

minhkiwi14 committed:
Add support for save

minhkiwi14 committed:
Merge branch 'branch-Level-7'

minhkiwi14 committed:
Modify class Deadline to add support for date and time

minhkiwi14 committed:
Add support for date and time

minhkiwi14 committed:
Merge branch 'branch-Level-8'

minhkiwi14 committed:
Add skeleton for Ui, Parser and TaskList

minhkiwi14 committed:
Implement class TaskList

minhkiwi14 committed:
Implement class Ui

minhkiwi14 committed:
Re-implement class Storage and others

minhkiwi14 committed:
Implement class Parser

minhkiwi14 committed:
Modify class Kiwi

minhkiwi14 committed:
Merge branch 'add-gradle-support'

minhkiwi14 committed:
Divide classes into packages

minhkiwi14 committed:
Edit README.md

minhkiwi14 committed:
Edit class name in Gradle

minhkiwi14 committed:
Add tests for classes Parser and Event

minhkiwi14 committed:
Edit archive name for the JAR file

minhkiwi14 committed:
Add JavaDoc comments for all public classes and methods

minhkiwi14 committed:
Follow coding standards

minhkiwi14 committed:
Add support for find

minhkiwi14 committed:
Merge branch 'branch-A-JavaDoc'

minhkiwi14 committed:
Merge branch 'branch-A-CodingStandard'

minhkiwi14 committed:
Merge conflicts

SociallyIneptWeeb commented on src/main/java/kiwi/command/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+
+import java.util.ArrayList;
+
+/**
+ * Manages a list of {@link Task} objects, providing methods to manipulate and access tasks.
+ * Supports adding, removing, marking, unmarking, and retrieving tasks by index.
+ */
```
Great work going out of your way to add links and other symbols in the Javadoc comments!

minhkiwi14 replied:
Thank you!

SociallyIneptWeeb commented on src/main/java/kiwi/task/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+        String[] dateTimeParts = by.split(" ", 2);
+        this.date = LocalDate.parse(dateTimeParts[0]);
+
+        // Solution adapted from ChatGPT with the prompt:
+        // "How to handle the case where time is missing?"
+        this.time = dateTimeParts.length > 1
+                ? LocalTime.parse(dateTimeParts[1])
+                : LocalTime.of(23, 59);
```
Cool use of ChatGPT to help out with something that you were unsure of how to handle!

minhkiwi14 replied:
Thank you very much!

SociallyIneptWeeb commented on src/main/java/kiwi/task/Task.java:
```diff
@@ -0,0 +1,56 @@
+package kiwi.task;
+
+/**
+ * Represents a generic task with a description and completion status.
+ * Serves as the base class for specific task types (Todo, Deadline, Event).
+ */
+public class Task {
```
Maybe you can convert this class into an abstract class?

minhkiwi14 replied:
I am considering this direction. Thank you for your recommendation.

SociallyIneptWeeb reviewed (commented):
Great work! Couldn't find any glaring coding standard or code quality guidelines being violated!

minhkiwi14 committed:
Add the Checkstyle support

minhkiwi14 committed:
Merge branch 'branch-A-CheckStyle'

hengjeekuan commented on src/main/java/kiwi/ui/Ui.java:
```diff
@@ -0,0 +1,141 @@
+/**
+ * The Ui class handles all user interactions, such as displaying messages and errors,
+ * and printing tasks in the list.
+ */
```
Good job on writing header comments for all your classes and methods.

minhkiwi14 replied:
Thank you!

hengjeekuan commented on src/main/java/kiwi/command/Parser.java:
```diff
@@ -0,0 +1,73 @@
+/**
+ * Provides utility methods for parsing user input arguments into specific components required by the application.
+ */
+public class Parser {
```
LGTM

minhkiwi14 replied:
Thank you!

hengjeekuan commented on src/main/java/kiwi/command/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+     * Finds tasks containing the specified keyword in their description.
+     * The search is case-insensitive.
+     *
+     * @param keyword The search term to match
```
LGTM, nothing to nitpick at all, just missing a fullstop here.

minhkiwi14 replied:
Oh I forgot that, I will fix soon. Thank you!

hengjeekuan commented on src/main/java/kiwi/storage/Storage.java:
```diff
@@ -0,0 +1,154 @@
+ * Supports loading tasks from a predefined file format and saving tasks in a format
+ * suitable for later reconstruction.
+ */
+public class Storage {
```
LGTM

minhkiwi14 replied:
Thank you!

hengjeekuan commented on src/main/java/kiwi/task/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+ * Formats deadlines as "MMM d yyyy" (e.g., Oct 10 2023) and times as "h a" (e.g., 3PM) in output.
+ * Inherits base task functionality from {@link Task}.
+ */
+public class Deadline extends Task {
```
LGTM

minhkiwi14 replied:
Thank you!

hengjeekuan commented on src/main/java/kiwi/task/Event.java:
```diff
@@ -0,0 +1,55 @@
+ * Inherits base task functionality from {@link Task}.
+ * Events are stored and displayed with user-provided time descriptors without parsing.
+ */
+public class Event extends Task {
```
LGTM

minhkiwi14 replied:
Thank you!

minhkiwi14 committed:
Add the skeleton for GUI classes

minhkiwi14 committed:
Add support for GUI

minhkiwi14 committed:
Merge branch 'branch-Level-10'

minhkiwi14 committed:
Enable assertions in Gradle

minhkiwi14 committed:
Add some assertions

minhkiwi14 committed:
Remove a redundant comment

minhkiwi14 committed:
Refactor the class Storage

The Storage class is refactored to improve code quality and make it easier to maintain.

This includes removing magic numbers and replacing them with well-named constants, ensuring consistency throughout the code. Additionally, error handling has been enhanced with more descriptive messages for loading and saving tasks, making debugging easier in case of file operation failures.

To improve readability, the task parsing logic has been simplified by breaking down complex methods into smaller, more focused helper methods. This adheres to the SLAP principle and helps make the code more modular and easier to understand.

minhkiwi14 committed:
Use Java Streams

Traditional for-loops in printTasks() and showFoundTasks() methods are
replaced with streams to improve code conciseness and readability.

Key changes:
* Use IntStream.range() with mapToObj() to generate indexed task strings
* Handle checked KiwiException in stream lambdas using try-catch blocks
* Maintain original exception propagation behavior by unwrapping
RuntimeExceptions

minhkiwi14 committed:
Merge pull request #1 from minhkiwi14/branch-A-Assertions

Branch a assertions

minhkiwi14 committed:
Merge branch 'master' into branch-A-CodeQuality

minhkiwi14 committed:
Merge branch 'master' into branch-A-Streams

minhkiwi14 committed:
Merge pull request #2 from minhkiwi14/branch-A-CodeQuality

Branch a code quality

minhkiwi14 committed:
Merge branch 'master' into branch-A-Streams

minhkiwi14 committed:
Merge pull request #3 from minhkiwi14/branch-A-Streams

Use Java Streams

minhkiwi14 committed:
Change some imports

minhkiwi14 committed:
Add some support for edit

minhkiwi14 committed:
Change messages

minhkiwi14 committed:
Merge branch 'branch-BCD-Extension'

minhkiwi14 committed:
Implement the Update extension

minhkiwi14 committed:
Rename boolean variables

minhkiwi14 committed:
Add Javadoc comments

minhkiwi14 committed:
Make the chatbot window fixed when resizing

minhkiwi14 committed:
Add product image

minhkiwi14 committed:
Update the User Guide

minhkiwi14 committed:
Fix bug for Storage

minhkiwi14 committed:
Update README.md

minhkiwi14 committed:
Follow the coding standard

minhkiwi14 committed:
Fix bugs after smoke testing

minhkiwi14 committed:
Add support for delayed exiting

minhkiwi14 committed:
Add credits

minhkiwi14 committed:
Update the User Guide

minhkiwi14 committed:
Make minor modifications to the UG
